<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
  <head>
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <meta name="author" content="Linghua Zhang" />
    <title>fossil HTTPS patch</title>
    <link rel="shortcut icon" href="/favicon.ico">
    <link href="/atom.xml" rel="alternate" title="setImpl" type="application/atom+xml" />
    <link rel="stylesheet" href="/media/css/style.css">
    <link rel="stylesheet" href="/media/css/github.css">
    <script type="text/javascript" src="/media/js/highlight.pack.js"></script>
    <script type="text/javascript">
      hljs.initHighlightingOnLoad();
      </script>
    </head>
    <body>
      <div id="container">
        <div id="main" role="main">
          <header>
          <h1>fossil HTTPS patch</h1>
          </header>
          <nav>
          <span><a title="home page" class="" href="/">home</a></span>
          <span><a title="tags" class="" href="/tags.html">tags</a></span>
          <span><a title="about" class="" href="/about.html">about me</a></span>
          <span><a title="my photos" class="" href="http://www.flickr.com/photos/lhzhang">gallery</a></span>
          <span><a title="blogroll" class="" href="/links.html">links</a></span>
          <span><a title="subscribe me" class="" href="/atom.xml">feed</a></span>
          </nav>
          <article class="content">
          <section class="post">
<p>最近用fossil搭了个wiki，不过原版尽管支持https schema的url生成，却漏掉了https header的判断，小小patch了一下</p>

<pre><code>--- src/cgi.c   2010-07-10 01:57:40.000000000 +0900
+++ src/cgi.c   2010-07-21 16:22:41.000000000 +0900
@@ -1140,6 +1140,8 @@
       cgi_setenv("HTTP_IF_NONE_MATCH", zVal);
     }else if( strcmp(zFieldName,"if-modified-since:")==0 ){
       cgi_setenv("HTTP_IF_MODIFIED_SINCE", zVal);
+    }else if( strcmp(zFieldName,"https:")==0 ){
+      cgi_setenv("HTTPS", zVal);
     }
   }
</code></pre>

<p>至于fossil是个什么东西以后有空整理一下，感兴趣的童鞋可以去 <a href="http://www.fossil-scm.org">Fossil</a> 看下</p>

</section>
<section class="meta">

<span class="tags">
  tagged by 
  
  <a href="/tags.html#fossil" title="fossil">fossil</a>&nbsp;
  
</span>

<span class="time">
  <time datetime="2010-07-30">2010-07-30</time>
</span>
</section>

<section class="comment">
<div id="disqus_thread"></div>
<script type="text/javascript">
    /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
    var disqus_shortname = 'gopherwood'; // required: replace example with your forum shortname
	var disqus_identifier = 'urn:uuid:e14a2804-f4f2-4037-b1b2-72129894167d';

    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = 'http://' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>


          </article>
        </div>
      </div> <!--! end of #container -->
    </body>
  </html>
